### PR TITLE
Enable Push on Android < 5.0 (less then API 21)

### PR DESCRIPTION
**Pull request by Hitabis**

### CODE REVIEW

**Comment by Hitabis** [comment]
On older android version the foreground property was always returning true even if the app was running in background.

**Comment by npomfret** [comment]
Hi - apologies for the delay, and thank you for submitting this change.  Struggling to find time to look at PR's on this project. 

Can you please make sure you describe the motivation for this change, and crucially **what is the expected change in behaviour**?  How could someone test it?

I would suggest you ask the authors of other outstanding PR's to test your change in a private fork.  If you can get the thumbs up from a couple of other developers then I'd be happy to merge it.  In exchange you could offer to test theirs maybe..?  That would really help get this project moving.

**Comment by yianyue** [comment]
For us, this PR fixed a bug for Android sdk < 21 where notifications do not show up when app is in the background but not closed. The notification object had `foreground = true`.

We did need to add `<uses-permission android:name="android.permission.GET_TASKS" android:maxSdkVersion="20" />` to the android manifest.

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationListenerService.java [hunk 1]
```diff
@@ -115,7 +123,12 @@ private void handleRemotePushNotification(ReactApplicationContext context, Bundl
     }
 
     private boolean isApplicationInForeground() {
+        if(VERSION.SDK_INT < 21){
+            return appInForegroundForOldApi();
+        }
         ActivityManager activityManager = (ActivityManager) this.getSystemService(ACTIVITY_SERVICE);
+
+        List<ActivityManager.RunningServiceInfo> serviceInfos = activityManager.getRunningServices(Integer.MAX_VALUE);
```
Comment (varungupta85) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationListenerService.java]: You are not using `serviceInfos` anywhere. Is this required?

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationListenerService.java [hunk 2]
```diff
@@ -7,6 +7,11 @@
 import android.os.Handler;
 import android.os.Looper;
 import android.util.Log;
+import android.os.Build.VERSION;
+import android.app.usage.UsageStatsManager;
+import android.app.usage.UsageStats;
+import android.content.Context;
+
```
Comment (varungupta85) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationListenerService.java]: I don't think any of the extra imports are also required except `import android.os.Build.VERSION;`.

**Pull request closed without merging**